index.js

```diff
@@ -598,7 +598,7 @@ exports.isValidSignature = function (v, r, s, homestead, chainId) {
     return false
   }
 
-  if ((homestead === false) && (new BN(s).cmp(SECP256K1_N_DIV_2) === 1)) {
+  if (homestead && (new BN(s).cmp(SECP256K1_N_DIV_2) === 1)) {
```

Comment:
Staring at this for some time now. Actually in the docs it is noted that ``homestead`` is set to ``true`` by default, however this is enforced nowhere in the code. My theory that this went unnoticed is the following: the ``chainId`` introduction is pretty new, just lately with the [v6.0.0](https://github.com/ethereumjs/ethereumjs-util/releases/tag/v6.0.0) release. I would assume that people before just have used this without setting ``homestead`` explicitly, assuming that the docs comment is correct. By having this check inverted this was then actually acting in a homestead way by a two-false-boolean double flip.

I would therefore suggest that we do this double change and add the ``homestead`` default setting along with this, I would assume this should play out in combination. 😄

Does this make sense? If you go along just update the PR or otherwise note where you disagree.

Reply:
I agree with your above assessment and updated the code accordingly 777241f

I also renamed the `homestead` param to `homesteadOrLater` as the validation rule is meant to apply to homestead and all later forks.